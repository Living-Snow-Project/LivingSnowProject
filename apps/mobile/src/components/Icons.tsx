import React from "react";
import { Platform, ViewStyle, StyleProp } from "react-native";
import * as Icon from "@expo/vector-icons";
import Colors from "../constants/Colors";
import { isSample } from "../record/Record";
import { AlgaeRecordTypePropType } from "../record/PropTypes";
import TestIds from "../constants/TestIds";

function StockIcon(props: StockIconProps) {
  const { name, size, color, style, testID } = props;
  return (
    <Icon.Ionicons
      name={name}
      size={size}
      style={style}
      color={color}
      testID={testID}
    />
  );
}

type IconNameType = keyof typeof Icon.Ionicons.glyphMap;

type StockIconProps = {
  name: IconNameType;
  size?: number;
  style?: StyleProp<ViewStyle>;
  color?: string;
  testID: string;
};

type SnowIconProps = {
  style?: StyleProp<ViewStyle>;
};

StockIcon.defaultProps = {
  size: 32,
  style: undefined,
  color: Colors.tabIconDefault,
};

function RecordIcon({ type }: { type: AlgaeRecordType }) {
  let name: IconNameType = Platform.OS === "ios" ? "ios-eye" : "md-eye";
  let color = "skyblue";
  let testID = TestIds.Icons.SightingIcon;

  if (isSample(type)) {
    name = Platform.OS === "ios" ? "ios-flask" : "md-flask";
    color = "lightcoral";
    testID = TestIds.Icons.SampleIcon;
  }

  return <StockIcon name={name} size={48} color={color} testID={testID} />;
}

RecordIcon.propTypes = {
  type: AlgaeRecordTypePropType.isRequired,
};

function PictureIcon() {
  return (
    <StockIcon
      name={Platform.OS === "ios" ? "ios-image" : "md-image"}
      color="green"
      size={48}
      testID={TestIds.Icons.PictureIcon}
    />
  );
}

function DeleteIcon() {
  return (
    <StockIcon
      name={Platform.OS === "ios" ? "ios-trash-outline" : "md-trash-outline"}
      color="darkred"
      testID={TestIds.Icons.DeleteIcon}
    />
  );
}

function EditIcon() {
  return (
    <StockIcon
      name={Platform.OS === "ios" ? "ios-open-outline" : "md-open-outline"}
      color="green"
      testID={TestIds.Icons.EditIcon}
    />
  );
}

function ScrollTopIcon() {
  return (
    <StockIcon
      size={48}
      name={
        Platform.OS === "ios"
          ? "ios-chevron-up-circle-outline"
          : "md-chevron-up-circle-outline"
      }
      color="black"
      testID={TestIds.Icons.ScrollTopIcon}
    />
  );
}

function SnowIcon(props: SnowIconProps) {
  const { style } = props;
  return (
    <StockIcon
      style={style}
      name={Platform.OS === "ios" ? "ios-snow" : "md-snow"}
      testID={TestIds.Icons.SnowIcon}
    />
  );
}

SnowIcon.defaultProps = {
  style: {},
};

export {
  EditIcon,
  DeleteIcon,
  PictureIcon,
  RecordIcon,
  ScrollTopIcon,
  SnowIcon,
  StockIcon,
};
