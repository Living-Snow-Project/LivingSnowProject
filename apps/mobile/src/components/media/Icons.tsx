import React from "react";
import { Ionicons } from "@expo/vector-icons";
import { AddIcon, Icon, useColorModeValue } from "native-base";
import { Colors, TestIds } from "../../constants";

type StockIconProps = {
  name: IconNameType;
  size?: number;
  color?: string;
  testID: string;
};

export function StockIcon({
  name,
  size = 32,
  color = Colors.tabIconDefault,
  testID,
}: StockIconProps) {
  return <Ionicons name={name} size={size} color={color} testID={testID} />;
}

type IconNameType = keyof typeof Ionicons.glyphMap;

export function AddPhotosIcon() {
  const color = useColorModeValue("primary.600", "primary.400");

  return (
    <AddIcon color={color} size="50" testID={TestIds.Icons.AddPhotosIcon} />
  );
}
export function PictureIcon() {
  return (
    <Icon
      as={Ionicons}
      color="green"
      name="image"
      size="xl"
      testID={TestIds.Icons.PictureIcon}
    />
  );
}

export function SnowIcon() {
  return (
    <Icon
      as={Ionicons}
      color="pink.500"
      name="snow"
      size="xl"
      testID={TestIds.Icons.SnowIcon}
    />
  );
}
