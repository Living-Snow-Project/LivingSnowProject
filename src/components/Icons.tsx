import React from "react";
import { Platform, ViewPropTypes } from "react-native";
import PropTypes from "prop-types";
import * as Icon from "@expo/vector-icons";
import Colors from "../constants/Colors";
import { isSample } from "../record/Record";
import { AlgaeRecordTypePropType } from "../record/PropTypes";
import TestIds from "../constants/TestIds";

function StockIcon({ name, size, style, color, testID }) {
  return (
    <Icon.Ionicons
      name={name}
      size={size}
      style={style}
      color={color}
      testID={testID}
    />
  );
}

StockIcon.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.number,
  style: ViewPropTypes.style,
  color: PropTypes.string,
  testID: PropTypes.string.isRequired,
};

StockIcon.defaultProps = {
  size: 32,
  style: undefined,
  color: Colors.tabIconDefault,
};

function RecordIcon({ type }: { type: AlgaeRecordType }) {
  let name = Platform.OS === "ios" ? "ios-eye" : "md-eye";
  let color = "skyblue";
  let testID = TestIds.Icons.SightingIcon;

  if (isSample(type)) {
    name = Platform.OS === "ios" ? "ios-flask" : "md-flask";
    color = "lightcoral";
    testID = TestIds.Icons.SampleIcon;
  }

  return <StockIcon name={name} size={48} color={color} testID={testID} />;
}

RecordIcon.propTypes = {
  type: AlgaeRecordTypePropType.isRequired,
};

function PictureIcon() {
  return (
    <StockIcon
      name={Platform.OS === "ios" ? "ios-image" : "md-image"}
      color="green"
      size={48}
      testID={TestIds.Icons.PictureIcon}
    />
  );
}

function SnowIcon({ style }) {
  return (
    <StockIcon
      style={style}
      name={Platform.OS === "ios" ? "ios-snow" : "md-snow"}
      testID={TestIds.Icons.SnowIcon}
    />
  );
}

SnowIcon.propTypes = {
  style: ViewPropTypes.style,
};

SnowIcon.defaultProps = {
  style: {},
};

export { PictureIcon, RecordIcon, StockIcon, SnowIcon };
