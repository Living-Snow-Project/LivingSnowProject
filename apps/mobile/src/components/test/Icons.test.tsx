import React from "react";
import { Platform } from "react-native";
import { render } from "@testing-library/react-native";
import { NativeBaseProviderForTesting } from "../../../jesttest.setup";
import { PictureIcon, SnowIcon } from "../Icons";

describe("Icons test suite", () => {
  describe("android", () => {
    beforeEach(() => {
      Platform.OS = "android";
    });

    afterEach(() => {
      Platform.OS = "ios";
    });

    test("picture icon", () => {
      const { toJSON } = render(
        <NativeBaseProviderForTesting>
          <PictureIcon />
        </NativeBaseProviderForTesting>
      );
      expect(toJSON()).toMatchSnapshot();
    });

    test("snow icon", () => {
      const { toJSON } = render(
        <NativeBaseProviderForTesting>
          <SnowIcon />
        </NativeBaseProviderForTesting>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe("ios", () => {
    test("picture icon", () => {
      const { toJSON } = render(
        <NativeBaseProviderForTesting>
          <PictureIcon />
        </NativeBaseProviderForTesting>
      );
      expect(toJSON()).toMatchSnapshot();
    });

    test("snow icon", () => {
      const { toJSON } = render(
        <NativeBaseProviderForTesting>
          <SnowIcon />
        </NativeBaseProviderForTesting>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
