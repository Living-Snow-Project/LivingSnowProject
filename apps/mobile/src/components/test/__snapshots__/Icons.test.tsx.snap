// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons test suite android picture icon 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text />
</RNCSafeAreaProvider>
`;

exports[`Icons test suite android snow icon 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text />
</RNCSafeAreaProvider>
`;

exports[`Icons test suite ios picture icon 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text />
</RNCSafeAreaProvider>
`;

exports[`Icons test suite ios snow icon 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text />
</RNCSafeAreaProvider>
`;
